dataplane/globalbase.cpp

```diff
@@ -1702,10 +1703,18 @@ balancer_real_id_t* generation::rebuild_service_ring_one_chash(
 		ipv6_address_t& ip = balancer_reals[real_id].destination;
 		reals.emplace_back(ip, real_id);
 	}
-	auto updater = chash::WeightUpdater::MakeWeightUpdater(reals, service->details->segments_per_real);
-	updater.InitLookup(ServiceWeights(service), start);
+	auto updater = chash::WeightUpdater::MakeWeightUpdater(
+	        reals,
+	        service->details->siderings_count,
+	        service->details->segments_per_weight);
+	if (!updater)
+	{
+		YANET_LOG_ERROR("Failed to intialize updater for balancer service");
+		std::abort();
```

Comment:
here we can use `YANET_THROW`. We have exceptions disabled, so it will expand to `YANET_LOG_ERROR` + `std::abort()` anyway:
```cpp
#if __cpp_exceptions
#define YANET_THROW(message) throw std::runtime_error(std::string(message))
#else // __cpp_exceptions
#define YANET_THROW(message)                                               \
	do                                                                 \
	{                                                                  \
		YANET_LOG_ERROR("%s\n", std::string_view(message).data()); \
		std::abort();                                              \
	} while (0)
#endif // __cpp_exceptions
```